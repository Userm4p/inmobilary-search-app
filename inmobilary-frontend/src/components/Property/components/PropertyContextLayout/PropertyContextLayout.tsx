import { PropertyContext } from '@/context/PropertyContext';
import { useProperty } from '@/hooks/useProperty';
import { useParams } from 'next/navigation';
import React from 'react';

interface Props {
  children: React.ReactNode;
}

export const PropertyContextLayout = ({ children }: Props) => {
  const { id } = useParams();
  const value = useProperty(id as string);

  return (
    <>
      <PropertyContext.Provider value={value}>{children}</PropertyContext.Provider>
    </>
  );
};
