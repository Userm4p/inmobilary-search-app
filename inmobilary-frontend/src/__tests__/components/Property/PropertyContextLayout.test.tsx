import { render } from '@testing-library/react';
import { PropertyContextLayout } from '@/components/Property/components/PropertyContextLayout/PropertyContextLayout';

jest.mock('next/navigation', () => ({
  useParams: jest.fn(() => ({ id: '123' })),
}));

describe('PropertyContextLayout', () => {
  it('should render children wrapped in PropertyContext provider', () => {
    const { container } = render(
      <PropertyContextLayout>
        <div data-testid="test-child">Test Child</div>
      </PropertyContextLayout>
    );

    expect(container).toMatchSnapshot();
  });
});
