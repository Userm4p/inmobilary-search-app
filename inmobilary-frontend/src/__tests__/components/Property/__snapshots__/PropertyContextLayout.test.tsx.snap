// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`PropertyContextLayout should render children wrapped in PropertyContext provider 1`] = `
<div>
  <div
    data-testid="test-child"
  >
    Test Child
  </div>
</div>
`;
